Sources/AppKitNavigation/Navigation/ModalSessionContent.swift

```diff
@@ -0,0 +1,22 @@
+#if canImport(AppKit) && !targetEnvironment(macCatalyst)
+
+import AppKit
+
+@MainActor
+public protocol ModalSessionContent: ModalContent {
+    func appKitNavigationBeginModalSession() -> NSApplication.ModalSession
```

Comment:
Do you think it's possible to avoid these protocols? In UIKitNavigation we got by using the classes themselves, but maybe I'm missing something?

Reply:
The controls that can perform ModalSession are not only NSWindow, but also NSAlert, NSSavePanel, etc. The same applies to other Sheet protocols.

Reply:
There are a few comments about public protocols that might be a little excessive for the public API, if they greatly simplify things internally I think there might be an option to hide them using `@_spi` 🤔

Reply:
@maximkrouk I tried it and found that it doesn't work. If you use this protocol for spi, all methods that use this protocol need to be spi-compatible.